admin.go

```diff
@@ -167,6 +170,27 @@ func handleLoadConfig(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func handleStop(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
+		return
+	}
+	log.Println("[ADMIN] Initiating shutdown")
+	if err := stopandCleanup(); err != nil {
+		log.Printf("[ADMIN][ERROR] stopping: %v \n", err)
+	}
+	log.Println("[ADMIN] Exiting")
+	os.Exit(0)
+}
+
+func stopandCleanup() error {
```

Comment:
`stopAndCleanup()`

And perhaps this should also be called by `gracefulStop()` here: https://sourcegraph.com/github.com/caddyserver/caddy@b44a22a9d45443aa736a04495dfe716b62b58adc/-/blob/sigtrap.go#L61-68 (we'd want to preserve the logic but reuse the code which calls `Stop()` then `CleanUpOwnLocks()`) - in order to reduce code duplication.

Reply:
done